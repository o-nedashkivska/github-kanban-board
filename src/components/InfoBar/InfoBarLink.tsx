import { Button } from "antd";

interface InfoBarLinkProps {
  text: string;
  href: string;
}

const InfoBarLink: React.FC<InfoBarLinkProps> = ({
  text,
  href,
}: InfoBarLinkProps) => {
  const upCaseFirst = text[0].toUpperCase() + text.slice(1);

  return (
    <Button type="link" size="large" target="_blank" href={href}>
      {upCaseFirst}
    </Button>
  );
};

export default InfoBarLink;
