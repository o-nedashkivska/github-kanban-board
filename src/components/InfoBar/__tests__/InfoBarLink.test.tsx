import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";

import InfoBarLink from "../InfoBarLink";

describe("InfoBarLink", () => {
  const text = "facebook";
  const href = "https://github.com/facebook";

  test("displays link with correct text and href", () => {
    render(<InfoBarLink text={text} href={href} />);

    expect(screen.getByText("Facebook")).toBeInTheDocument();

    expect(screen.getByRole("link")).toBeInTheDocument();
    expect(screen.getByRole("link")).toHaveAttribute(
      "href",
      "https://github.com/facebook"
    );
    expect(screen.getByRole("link")).toHaveAttribute("target", "_blank");
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<InfoBarLink text={text} href={href} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
