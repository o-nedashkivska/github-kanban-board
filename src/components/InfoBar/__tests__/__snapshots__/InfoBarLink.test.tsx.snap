// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBarLink matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="ant-btn css-dev-only-do-not-override-1kuana8 ant-btn-link ant-btn-lg"
    href="https://github.com/facebook"
    tabindex="0"
    target="_blank"
  >
    <span>
      Facebook
    </span>
  </a>
</DocumentFragment>
`;
